src/Cake.Scripting.Transport/Serialization/FileChangeSerializer.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.IO;
+using Cake.Scripting.Abstractions.Models;
+
+namespace Cake.Scripting.Transport.Serialization
+{
+    internal static class FileChangeSerializer
+    {
+        public static void Serialize(BinaryWriter writer, FileChange fileChange)
+        {
+            if (writer == null)
+            {
+                throw new ArgumentNullException(nameof(writer));
+            }
+            if (fileChange == null)
+            {
+                throw new ArgumentNullException(nameof(fileChange));
+            }
+
```

Comment:
Perhaps write an identifier for the type being serialized as well as a version (`byte` or `ushort`)?